libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -64,32 +64,45 @@ AP_GPS_SBF::AP_GPS_SBF(AP_GPS &_gps, AP_GPS::GPS_State &_state,
 bool
 AP_GPS_SBF::read(void)
 {
-    uint32_t now = AP_HAL::millis();
-
-    if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE &&
-        _init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
-        const char *init_str = _initialisation_blob[_init_blob_index];
-
-        if (now > _init_blob_time) {
-            if (now > _config_last_ack_time + 2500) {
-                // try to enable input on the GPS port if we have not made progress on configuring it
-                Debug("SBF Sending port enable");
-                port->write((const uint8_t*)_port_enable, strlen(_port_enable));
-                _config_last_ack_time = now;
-            } else {
-                Debug("SBF sending init string: %s", init_str);
-                port->write((const uint8_t*)init_str, strlen(init_str));
-            }
-            _init_blob_time = now + 1000;
-        }
-    }
-
     bool ret = false;
-    while (port->available() > 0) {
+    uint32_t available_bytes = port->available();
+    for (uint32_t i = 0; i < available_bytes; i++) {
         uint8_t temp = port->read();
         ret |= parse(temp);
     }
 
+    if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE) {
+        if (_init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
+            uint32_t now = AP_HAL::millis();
+            const char *init_str = _initialisation_blob[_init_blob_index];
+
+            if (now > _init_blob_time) {
+                if (now > _config_last_ack_time + 2500) {
+                    // try to enable input on the GPS port if we have not made progress on configuring it
+                    Debug("SBF Sending port enable");
+                    port->write((const uint8_t*)_port_enable, strlen(_port_enable));
+                    _config_last_ack_time = now;
+                } else {
+                    Debug("SBF sending init string: %s", init_str);
+                    port->write((const uint8_t*)init_str, strlen(init_str));
+                }
+                _init_blob_time = now + 1000;
+            }
+        } else if (gps._raw_data >= 2) { // only manage disarm/rearms when the user opts into it
+            if (hal.util->get_soft_armed()) {
+                _has_been_armed = true;
+            } else if (_has_been_armed && (RxState & SBF_DISK_MOUNTED)) {
+                // unmounting the disk will be rate limited with the init blob timer
+                // since we have finished init, the init blob timer is now fre, the init blob timer is now freee
```

Comment:
Maybe it's me, but I don't understand anything that is written here 😃

Reply:
Other then a terrible misappropriation of `e`'s basically I need a timer to rate limit disarming, otherwise we spam the GPS aggressively for up to a second with unmount commands. We should have only gotten here if the configuration has passed (or arming checks are disabled). So rather then creating another 32 bit field to store the last unmount attempt time in, it was more efficient to recycle the initial config sending one.

Reply:
Oh boy, so many words to describe something so simple: `since init is done at this point and we want unmounting to be rate limited, we take over the _init_blob_time variable` 😄

Reply:
I think the inadvertent insert in my original comment just killed it. Will fix :P